### PR TITLE
fstar: using unreleased z3 version 4.5.1.1f29cebd4df6

### PR DESCRIPTION
**Pull request by beurdouche**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/master/CONTRIBUTING.md)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `brew install <formula>`)?

-----

### CODE REVIEW

File: Formula/fstar.rb [hunk 1]
```diff
@@ -27,6 +31,13 @@ def install
     inreplace "src/ocaml-output/Makefile", "$(DATE_EXEC) -Iseconds",
                                            "$(DATE_EXEC) '+%Y-%m-%dT%H:%M:%S%z'"
 
+    resource("z3").stage do
+      system "python", "scripts/mk_make.py"
+      system "make", "-C", "build"
+
+      (libexec/"bin").install "build/z3"
```
Comment (ilovezfs) [inline Formula/fstar.rb]: It would be preferable to use `make install`

File: Formula/fstar.rb [hunk 2]
```diff
@@ -16,7 +16,11 @@ class Fstar < Formula
   depends_on "opam" => :build
   depends_on "gmp"
   depends_on "ocaml" => :recommended
-  depends_on "z3" => :recommended
+
```
Comment (ilovezfs) [inline Formula/fstar.rb]: would be good to add a comment about why we're using the special z3 instead of the z3 formula

Comment (beurdouche) [reply Formula/fstar.rb]: @ilovezfs Done

**Comment by ilovezfs** [comment]
Thanks @beurdouche! Merged :)

**Pull request closed without merging**